File: src/com/juhnowski/onthology/Picture.java
package com.juhnowski.onthology;

import com.juhnowski.onthology.logical.Situation;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.TreeSet;

public class Picture extends Model<Reality>{
    Situation situation;
    TreeSet<Element> picture;
    HashMap<Element,TheObject> elements;

    public class Element extends TheObject{

        public Element(ArrayList<Property> internal) {
            super(internal);
        }
    };

}

File: src/com/juhnowski/onthology/Picture.kt
package com.juhnowski.onthology

import com.juhnowski.onthology.logical.LogicalPicture
import com.juhnowski.onthology.logical.Situation

import java.util.ArrayList
import java.util.HashMap

abstract class Picture<T> : LogicalPicture<T>() {
    internal var pictorialForm: WorldForm<Element>? = null
    abstract fun depict(reality: T): ArrayList<AtomicFact>

}
